django/core/checks/security/base.py

```diff
@@ -235,6 +239,65 @@ def check_referrer_policy(app_configs, **kwargs):
     return []
 
 
+@register(Tags.security, deploy=True)
+def check_cache_and_media_root(app_configs, **kwargs):
+    if not settings.MEDIA_ROOT:
+        return []
+    _media_root = os.path.abspath(settings.MEDIA_ROOT)
+    for alias in settings.CACHES.keys():
+        if isinstance(caches[alias], FileBasedCache):
+            location_check = (
+                _media_root,
+                caches[alias]._dir,
+            )
+            if os.path.commonpath(location_check) == _media_root:
+                return [Warning(
+                    "Your cache configuration might expose or corrupt your file system.",
```

Comment:
This message confuses me. I can't see how my filesystem would become "corrupt".

Really we're talking about the cache being exposed in the media or static files. There is also a small risk of *overwriting data in `MEDIA_ROOT`* and/or the *cache being corrupted*.

Perhaps:

```suggestion
                    "Your configuration might expose your cache or lead to corruption of your data.",
```